front/components/assistant/conversation/MessageEdit.tsx

```diff
@@ -0,0 +1,130 @@
+import { Button } from "@dust-tt/sparkle";
+import type {
+  LightAgentConfigurationType,
+  MentionType,
+  UserMessageType,
+  WorkspaceType,
+} from "@dust-tt/types";
+import { EditorContent } from "@tiptap/react";
+import React, { useContext, useMemo } from "react";
+
+import useAssistantSuggestions from "@app/components/assistant/conversation/input_bar/editor/useAssistantSuggestions";
+import type { EditorMention } from "@app/components/assistant/conversation/input_bar/editor/useCustomEditor";
+import useCustomEditor, {
+  getJSONFromText,
+} from "@app/components/assistant/conversation/input_bar/editor/useCustomEditor";
+import { SendNotificationsContext } from "@app/components/sparkle/Notification";
+import { useSubmitFunction } from "@app/lib/client/utils";
+import { classNames } from "@app/lib/utils";
+
+interface MessageEdit {
+  conversationId: string;
+  owner: WorkspaceType;
+  userMessage: UserMessageType;
+  agentConfigurations: LightAgentConfigurationType[];
+  onClose: () => void;
+}
+
+export function MessageEdit({
+  conversationId,
+  owner,
+  userMessage,
+  agentConfigurations,
+  onClose,
+}: MessageEdit) {
+  const sendNotification = useContext(SendNotificationsContext);
+
+  const { submit: handleEditMessage, isSubmitting } = useSubmitFunction(
+    async (
+      isEmpty: boolean,
+      textAndMentions: { mentions: EditorMention[]; text: string }
+    ) => {
+      const { mentions: rawMentions, text } = textAndMentions;
+      const mentions: MentionType[] = rawMentions.map((m) => ({
+        configurationId: m.id,
+      }));
+
+      const body = {
+        content: text,
+        mentions,
+      };
+
+      const mRes = await fetch(
```

Comment:
We introduced the concept of optimistic UI to have a snappier UI (https://github.com/dust-tt/dust/pull/4410). Can we use this pattern here as well to update the user message in the conversation without waiting for the re-fetch of the conversation?

Reply:
@flvndvd I don't  see what to do here, the behavior seems to be the same when we post a new message and edit a message. Also I don't  clearly understand the changes in #4410 , if you can give me some hints about the pubsub mechanism ?

Reply:
It should be handled here : 8f059549fe60afc9fd5df9711f2fca300715c596